applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_geo_steady_state_piping_element.cpp

```diff
@@ -0,0 +1,293 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+#include "custom_elements/geo_steady_state_Pw_piping_element.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+#include "tests/cpp_tests/test_utilities.h"
+
+namespace
+{
+
+using namespace Kratos;
+
+PointerVector<Node> CreateNodes()
+{
+    PointerVector<Node> result;
+    result.push_back(Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0));
+    result.push_back(Kratos::make_intrusive<Node>(2, 1.0, 0.0, 0.0));
+    return result;
+}
+
+PointerVector<Node> CreateIdenticalNodes()
+{
+    PointerVector<Node> result;
+    result.push_back(Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0));
+    result.push_back(Kratos::make_intrusive<Node>(2, 0.0, 0.0, 0.0));
+    return result;
+}
+
+ModelPart& CreateModelPartWithWaterPressureVariableAndVolumeAcceleration(Model& rModel)
+{
+    auto& r_result = rModel.CreateModelPart("Main");
+    r_result.AddNodalSolutionStepVariable(WATER_PRESSURE);
+    r_result.AddNodalSolutionStepVariable(VOLUME_ACCELERATION);
+
+    return r_result;
+}
+
+GeoSteadyStatePwPipingElement<2, 2> CreateGeoSteadyStatePwPipingElementWithPWDofs(
+    const Properties::Pointer& rProperties, const Geometry<Node>::Pointer& rGeometry)
+{
+    auto result = GeoSteadyStatePwPipingElement<2, 2>{1, rGeometry, rProperties};
+    for (auto& node : result.GetGeometry()) {
+        node.AddDof(WATER_PRESSURE);
+    }
+
+    return result;
+}
+
+GeoSteadyStatePwPipingElement<2, 2> CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithPWDofs(
+    Model& rModel, const Properties::Pointer& rProperties)
+{
+    auto& r_model_part = CreateModelPartWithWaterPressureVariableAndVolumeAcceleration(rModel);
+
+    PointerVector<Node> nodes;
+    nodes.push_back(r_model_part.CreateNewNode(0, 0.0, 0.0, 0.0));
+    nodes.push_back(r_model_part.CreateNewNode(1, 1.0, 0.0, 0.0));
+    const auto p_geometry = std::make_shared<Line2D2<Node>>(nodes);
+    return CreateGeoSteadyStatePwPipingElementWithPWDofs(rProperties, p_geometry);
+}
+
+GeoSteadyStatePwPipingElement<2, 2> CreateHorizontalUnitLengthGeoSteadyStatePwPipingElementWithoutPWDofs(
+    Model& rModel, const Properties::Pointer& rProperties)
+{
+    auto& r_model_part = CreateModelPartWithWaterPressureVariableAndVolumeAcceleration(rModel);
+
+    PointerVector<Node> nodes;
+    nodes.push_back(r_model_part.CreateNewNode(1, 0.0, 0.0, 0.0));
+    nodes.push_back(r_model_part.CreateNewNode(2, 1.0, 0.0, 0.0));
+    const auto p_geometry = std::make_shared<Line2D2<Node>>(nodes);
```

Comment:
If you believe that the previous suggestion makes sense, then you may want to apply it here, too:
```suggestion
    const auto p_geometry = std::make_shared<Line2D2<Node>>(CreateNodes());
```

Reply:
done